### ISSUE TITLE
[Heist] Test mode

### ISSUE THREAD

**Issue by Brandjuh** [Redjumpman/Jumper-Plugins#112]
**Note: for support questions, please use the [Cog Support Server](https://discord.gg/c6HQUb7)**. This repository's issues are reserved for feature requests and bug reports.

# I'm submitting a ...
- [ ] bug report
- [X] feature request
- [ ] support request => Please do not submit support request here, see note at the top of this template.

# **Cog Information**

Name:  Heist
Version: -

# Feature Description

Test mode:
Option to start a heist with a X amount of persons to test the outcome of the heist and test the lines you have made.

I like to add lines to the heist and adjust some things regular. The only problem I have is that I can not test it. That why I would like see a option that you can set a amount of players and let the heist plays just like its normal. 

# Bug Report
N/A

**Command used:**
N/A
**Expected result:**
N/A
**Steps to reproduce:**
N/A
**Please tell me about your environment/OS:**
N/a

**Comment by skeith** [comment]
What if the test is not the whole heist but just the outcome line test?

So, outcome add/remove/edit feature and testing it

**Comment by Brandjuh** [comment]
Good question but I want test the lines and the outcome of the game. 

For the lines a command that print the lines would be fine.

For testing the location I prefer a more for "realistic" test, since that's the only way how you can feel/see how much people are dropping out and what the outcome is. 
Now I need to make a location test it with some test accounts to take a look of the outcome is what I want to have.

**Comment by skeith** [comment]
What would you wanted to achieve by testing the whole game? Isn't your main purpose was to see the newly added lines as outcome?

**Comment by Brandjuh** [comment]
To get the feeling how the games progress. I have a few heist themes that I use for several servers and the only way to test the "feeling" of it is by actually test it. Some locations are not worth to test with 2 users if they are made for 20. 

I am not able to get always that 20 persons. Still I want test it with that amount of players.

**Comment by Redjumpman** [comment]
When I rewrite heist, I will probably look at having a command run a test mode, letting you set parameters with the command like crew size. An example implementation could be:

`[p] heist test 15` and it will run through a game of heist using randomly generated users from your server. This will also be helpful for me as well, so it is something I definitely want to do.

The problem at the moment is that I do not want to add anything to cogs that work with Red V2, so feature will only come with the Red V3 implementation. Secondly, I have a lot of ideas and suggestions when it comes to rewriting heist as a whole. So I need to make sure the test encompasses those feature as well. Finally, Heist will be the last plugin I port over to V3. Mainly because it is such a massive code base, and I only see it growing with the rewrite.

So TL;DR yes I will do it. I also want it this feature. V3 only release. It will be quite sometime before I begin work on it.
